src/sdk/person/usePersonQuery.ts

```diff
@@ -0,0 +1,44 @@
+import { useEffect } from 'react'
+import { gql } from '@vtex/graphql-utils'
+import type {
+  PersonQueryQuery,
+  PersonQueryQueryVariables,
+} from '@generated/graphql'
+import { useSession } from '@faststore/sdk'
+
+import { useQuery } from '../graphql/useQuery'
+import type { QueryOptions } from '../graphql/useQuery'
+
+export const query = gql`
+  query PersonQuery {
+    person {
+      id
+      email
+      givenName
+      familyName
+      isAuthenticated
+    }
+  }
+`
+
+export const usePersonQuery = (options?: QueryOptions) => {
+  const { data } = useQuery<PersonQueryQuery, PersonQueryQueryVariables>(
+    query,
+    {},
+    { ...options, fetchOptions: { ...options?.fetchOptions, method: 'POST' } }
+  )
+
+  const { setSession, user, ...session } = useSession()
+  const person = data?.person
+
+  useEffect(() => {
+    if (person !== null && person !== undefined && person !== user) {
+      setSession({
+        ...session,
+        user: person,
+      })
+    }
+  })
```

Comment:
```suggestion
```suggestion
  useEffect(() => {
    if (person !== null && person !== undefined && person !== user) {
      setSession({
        ...session,
        user: person,
      })
    }
  }, [person, setSession, session])
```
```